### PR TITLE
Slack reporter

### PR DESCRIPTION
**Pull request by YuvalFeldman**

### CODE REVIEW

File: pkg/log/slackReporter.go [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+type MessageLevel string
+
+const (
+	InfoLog    MessageLevel = "info"
+	WarningLog MessageLevel = "warning"
+	ErrorLog   MessageLevel = "error"
+)
+
+func SendCloudReport(messageLevel MessageLevel, messageType string, payload *map[string]interface{}) error {
```
Comment (moshe-kabala) [inline pkg/log/slackReporter.go]: You never return an error on this function so you can remove the return error

File: pkg/log/slackReporter.go [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+
+	unameData := getUnameData()
+	message := map[string]interface{}{"messageType": messageType, "installId": cloudMessageUuid, "level": messageLevel,
+		"osName": runtime.GOOS, "payload": payload, "unameData": unameData}
+
+	jsonData, err := json.Marshal(message)
+	if err != nil {
+		Println("Error:", err)
```
Comment (moshe-kabala) [inline pkg/log/slackReporter.go:51]: Remider use VerboseLogger

File: pkg/log/slackReporter.go [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+	req, _ := http.NewRequest("POST", reportURL, bytes.NewBuffer(jsonData))
+	req.Header.Add("Content-Type", "application/json")
+
+	res, err := http.DefaultClient.Do(req)
+	if err != nil {
+		Printf("POST request failed: %v\n", err)
+		return nil
+	}
+	if res.StatusCode != http.StatusOK {
+		Printf("POST request failed with status code: %d\n", res.StatusCode)
```
Comment (moshe-kabala) [inline pkg/log/slackReporter.go:64]: Remider use VerboseLogger

File: cmd/local/install.go [hunk 4]
```diff
@@ -8,6 +8,7 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/tensorleap/cli-go/pkg/k3d"
 	"github.com/tensorleap/cli-go/pkg/local"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline cmd/local/install.go:11]: Just use log

File: pkg/helm/utils.go [hunk 5]
```diff
@@ -49,18 +51,23 @@ type HelmConfig struct {
 }
 
 func CreateHelmConfig(kubeContext, namespace string, logger *logrus.Logger) (*HelmConfig, error) {
+	tlLog.SendCloudReport("info", "install-script-create-helm", nil)
```
Comment (moshe-kabala) [inline pkg/helm/utils.go]: Move this to Install function on the helm

File: pkg/k3d/cluster.go [hunk 6]
```diff
@@ -17,6 +17,7 @@ import (
 	k3d "github.com/k3d-io/k3d/v5/pkg/types"
 	"github.com/k3d-io/k3d/v5/version"
 	"github.com/tensorleap/cli-go/pkg/log"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline pkg/k3d/cluster.go:20]: log

File: pkg/helm/upgrade.go [hunk 7]
```diff
@@ -3,15 +3,16 @@ package helm
 import (
 	"time"
 
-	"github.com/tensorleap/cli-go/pkg/log"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline pkg/helm/upgrade.go:6]: log

File: cmd/local/upgrade.go [hunk 8]
```diff
@@ -7,6 +7,7 @@ import (
 	"github.com/tensorleap/cli-go/pkg/helm"
 	"github.com/tensorleap/cli-go/pkg/k3d"
 	"github.com/tensorleap/cli-go/pkg/local"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline cmd/local/upgrade.go:10]: log

File: cmd/local/uninstall.go [hunk 9]
```diff
@@ -4,20 +4,22 @@ import (
 	"github.com/spf13/cobra"
 	"github.com/tensorleap/cli-go/pkg/k3d"
 	"github.com/tensorleap/cli-go/pkg/local"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline cmd/local/uninstall.go:7]: log

File: pkg/k3d/registry.go [hunk 10]
```diff
@@ -18,6 +18,7 @@ import (
 	docker "github.com/k3d-io/k3d/v5/pkg/runtimes/docker"
 	k3d "github.com/k3d-io/k3d/v5/pkg/types"
 	"github.com/tensorleap/cli-go/pkg/log"
+	tlLog "github.com/tensorleap/cli-go/pkg/log"
```
Comment (moshe-kabala) [inline pkg/k3d/registry.go:21]: log

**Review by moshe-kabala — APPROVED** [review]
LGMT

File: pkg/log/slackReporter.go (hunk 2 above)
Comment (YuvalFeldman) [reply pkg/log/slackReporter.go:52]: Updated

File: pkg/log/slackReporter.go (hunk 1 above)
Comment (YuvalFeldman) [reply pkg/log/slackReporter.go]: Updated

File: pkg/log/slackReporter.go (hunk 3 above)
Comment (YuvalFeldman) [reply pkg/log/slackReporter.go]: Updated

File: cmd/local/install.go (hunk 4 above)
Comment (YuvalFeldman) [reply cmd/local/install.go]: Updated

File: pkg/helm/utils.go (hunk 5 above)
Comment (YuvalFeldman) [reply pkg/helm/utils.go]: Updated

File: pkg/helm/upgrade.go (hunk 7 above)
Comment (YuvalFeldman) [reply pkg/helm/upgrade.go]: Updated

File: cmd/local/upgrade.go (hunk 8 above)
Comment (YuvalFeldman) [reply cmd/local/upgrade.go]: Updated

File: cmd/local/uninstall.go (hunk 9 above)
Comment (YuvalFeldman) [reply cmd/local/uninstall.go]: Updated

**Pull request merged**